### ISSUE TITLE
Inconsistency in model processor when checking property domain

### ISSUE THREAD

**Issue by kittaakos** [SemanticApplicationDesignLanguage/sadl#407]
@crapo, while working on #406 I ran into some inconsistency.

If I have the following in one SADL file:
```java
uri "http://sadl.org/x.sadl".
Shape is a class described by area with values of type float.
Rectangle is a type of Shape, described by height with values of type float, described by width with values of type float.
Circle is a type of Shape described by radius with values of type float.
Test: width of <|> // <-- I can see `Rectangle` here
```

Then I can see only `Rectangle` as a possible CA, as the property domain check is correct for the `Circle`: `:Circle is not in domain of property :width`.

However, when I split the models into individual files, `Circle` is considered to be valid.
`Shape.sadl`:
```java
uri "http://sadl.org/Shape.sadl".
Shape is a class described by area with values of type float.
```

`Circle.sadl`:
```java
uri "http://sadl.org/Circle.sadl".
import "http://sadl.org/Shape.sadl".
Circle is a type of Shape described by radius with values of type float.
```

`Rectangle.sadl`:
```java
uri "http://sadl.org/Rectangle.sadl".
import "http://sadl.org/Shape.sadl".
Rectangle is a type of Shape, described by height with values of type float, described by width with values of type float.
```

`Test.sadl`:
```java
uri "http://sadl.org/Test.sadl".
import "http://sadl.org/Circle.sadl".
import "http://sadl.org/Rectangle.sadl".
Test: width of <|> // <-- Besides `Rectangle`, `Circle` is also here :(
```

If you want me I can look into this, but I would expect you are way more familiar with the model processor and validator code.

Edit: `Shape` -> `Circle`.

**Comment by crapo** [comment]
I can look into it.

**Comment by crapo** [comment]
I see Circle in the first instance where everything is in one file. Have you made changes that would affect this? I don't recall where to set a breakpt to see what's happening here...

**Comment by kittaakos** [comment]
> Have you made changes that would affect this?

No.

**Comment by kittaakos** [comment]
> I don't recall where to set a breakpt to see what's happening here...

Here: https://github.com/crapo/sadlos2/blob/development/sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.jena/src/com/ge/research/sadl/jena/JenaBasedSadlModelProcessor.java#L903

**Comment by crapo** [comment]
The reason Circle is included for me in the single model case is because I have the preference set for type checking errors are warnings only and so in ProposalProviderValidationacceptor, none meet the Severity.ERROR filter criteria so Circle gets added.

**Comment by crapo** [comment]
When the model is spread across 4 files, and the validate call is for Circle was failing for a strange reason that I've fixed locally. I think you can go forward assuming for the moment that the multiple will behave, when fix is merged, like the single model. Even with the fix it doesn't work for the reason above--it's only a warning when preference says make warnings.

**Comment by crapo** [comment]
While there is much work to be done on content assist, and it will proceed under #601 , this modification fixes the issue in #407, which had to do with the preference setting to show type checking errors as warnings interfering with the filtering of content assist proposals. When PR #603 is merged this can be closed.
